backend/src/database/repositories/memberRepository.ts

```diff
@@ -273,18 +279,34 @@ class MemberRepository {
     })
   }
 
-  static async addToMerge(id, toMergeId, options: IRepositoryOptions) {
+  static async addToMerge(
+    suggestions: IMemberMergeSuggestion[],
+    options: IRepositoryOptions,
+  ): Promise<void> {
     const transaction = SequelizeRepository.getTransaction(options)
+    const seq = SequelizeRepository.getSequelize(options)
 
-    const returnPlain = false
-
-    const member = await this.findById(id, options, returnPlain)
+    // Process suggestions in chunks of 100 or less
+    const suggestionChunks = chunk(suggestions, 100)
 
-    const toMergeMember = await this.findById(toMergeId, options, returnPlain)
+    const insertValues = (memberId: string, toMergeId: string, similarity: number | null) =>
+      `('${memberId}', '${toMergeId}', ${!similarity ? 'NULL' : similarity}, NOW(), NOW())`
 
-    await member.addToMerge(toMergeMember, { transaction })
+    for (const suggestionChunk of suggestionChunks) {
+      const query = `
+      INSERT INTO "memberToMerge" ("memberId", "toMergeId", "similarity", "createdAt", "updatedAt")
+      VALUES ${suggestionChunk
+        .map((suggestion) =>
```

Comment:
this is not the best way to insert stuff - this can leave to SQL injection because we are building literal strings... If this is only for internal use OK but if there is a chance that this get's called from some user input/rest api call or smt like that we should really generate better SQL.

You have to generate insert like this to keep it safe:
```sql
insert into "memberToMerge" ("memberId", "toMergeId", ...)
values (:memberId0, :toMergeId0, ...), (:memberId1, :toMergeId1, ...), (:memberId2, :toMergeId2, ...);
```

and then pass all these via `replacements` parameter like so:
```typescript
await seq.query(query, {
  replacements: {
    memberId0: value,
    toMergeId0: value,
    memberId1: value,
    toMergeId1: value,
    ...
   },
   type: QueryTypes.INSERT,
   transaction,
})
```

Reply:
Thanks! In this case, this is not accessible through API. But fixing anyway just in case